### PR TITLE
parser, planner: fix embedded setOprStmt will be seen as SetOprSelectList item and lost its orderBy and Limit

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/49377

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
parser, planner: fix embedded setOprStmt will be seen as SetOprSelectList item and lost its orderBy and Limit
```

### CODE REVIEW

File: pkg/executor/test/executor/executor_test.go [hunk 1]
```diff
@@ -2724,3 +2724,36 @@ func TestProcessInfoOfSubQuery(t *testing.T) {
 	tk2.MustQuery("select 1 from information_schema.processlist where TxnStart != '' and info like 'select%sleep% from t%'").Check(testkit.Rows("1"))
 	wg.Wait()
 }
+
+func TestIssues49377(t *testing.T) {
```
Comment (bb7133) [inline pkg/executor/test/executor/executor_test.go:2728]: Could you change this to the MySQL-tester based test? （https://github.com/pingcap/tidb/tree/master/tests/integrationtest）

**Review by qw4990 — APPROVED** [review]

File: pkg/executor/test/executor/executor_test.go (hunk 1 above)
Comment (AilinKid) [reply pkg/executor/test/executor/executor_test.go:2728]: i‘ll leave it to next pr

**Review by tangenta — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**